city_scrapers/spiders/il_port_district.py

```diff
@@ -0,0 +1,183 @@
+import datetime
+import re
+
+import scrapy
+from city_scrapers_core.constants import BOARD, COMMITTEE, NOT_CLASSIFIED
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+
+
+class IlPortDistrictSpider(CityScrapersSpider):
+    name = "il_port_district"
+    agency = "Illinois International Port District"
+    timezone = "America/Chicago"
+    allowed_domains = ["www.iipd.com"]
+    location = {
+        "name": "Illinois International Port District ",
+        "address": "3600 E. 95th St. Chicago, IL 60617",
+    }
+
+    schedules_url = "https://www.iipd.com/calendar/schedules"
+
+    def start_requests(self):
+        yield scrapy.Request(
+            url="https://www.iipd.com/calendar/agendas", callback=self.parse_agendas
+        )
+        yield scrapy.Request(
+            url="https://www.iipd.com/about/board-meeting-minutes", callback=self.parse_minutes
+        )
+
+    @classmethod
+    def from_crawler(cls, crawler, *args, **kwargs):
+        spider = super().from_crawler(crawler, *args, **kwargs)
+        crawler.signals.connect(spider.spider_idle, signal=scrapy.signals.spider_idle)
+        return spider
+
+    def spider_idle(self):
+        """Call parse_schedules if spider is idle (finished parsing minutes and agendas)"""
+        self.crawler.signals.disconnect(self.spider_idle, signal=scrapy.signals.spider_idle)
+        self.crawler.engine.crawl(
+            scrapy.Request(self.schedules_url, callback=self.parse_schedules), self
+        )
+        raise scrapy.exceptions.DontCloseSpider
+
+    def parse_schedules(self, response):
+
+        year = response.xpath("//em/text()").extract_first()[:4]
+
+        rows = response.xpath("//tr")
+        meeting_types = rows[0].xpath(".//strong/text()").extract()
+        meeting_types = [x.strip(" :s") for x in meeting_types]
+
+        strong_meetings = rows.xpath(".//strong/text()")[len(meeting_types):].extract()
+        if len(strong_meetings) % 2 != 0:
+            strong_meetings.append('')
+        strong_meetings = list(zip(strong_meetings[0::2], strong_meetings[1::2]))
+
+        additional_info = response.xpath("//p[contains(text(), '*')]/text()").extract()
+        changed_meeting_time = re.findall(r"\d{1,2}:\d{2}am|pm", additional_info[2])[0]
+        meeting_time = "9:00am"
+
+        self._validate_location(response)
+
+        for i, row in enumerate(strong_meetings + rows[1:]):
+            if i >= len(strong_meetings):
+                meetings_dates = row.xpath(".//div/text()").extract()
+                if not meetings_dates:
+                    continue
+            else:
+                meetings_dates = row
+
+            for i, date in enumerate(meetings_dates):
```

Comment:
It might make sense to put more of this date parsing logic in `_parse_start`. Also, the July 19th meeting is set as 8:30am and it looks like it shouldn't be.